src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -464,7 +464,14 @@ public async Task<HttpResponseMessage> SendWithRetryAsync(HttpRequestMessage req
 
                 try
                 {
-                    return await connection.SendAsync(request, doRequestAuth, cancellationToken).ConfigureAwait(false);
+                    if (connection is HttpConnection)
+                    {
+                        return await SendWithNtConnectionAuthAsync((HttpConnection)connection, request, doRequestAuth, cancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        return await connection.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                    }
```

Comment:
Nit: FWIW, code gen will end up being a bit better if this were instead written to have a single await instead of two, e.g.
```C#
Task<HttpResponseMessage> resp = connection is HttpConnection httpConnection ?
    SendWithNtConnectionAuthAsync(httpConnection, request, doRequestAuth, cancellationToken) :
    connection.SendAsync(request, cancellationToken);
return await resp.ConfigureAwait(false);
```